File: 0067-add-binary/0067-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int carry = 0;
        int i = a.size() - 1;
        int j = b.size() - 1;
        string result = "";
        
        while (i >= 0 || j >= 0 || carry) {
            int temp = 0;
            if (i >= 0 && a[i--] == '1') temp++;
            if (j >= 0 && b[j--] == '1') temp++;
            if (carry) temp++;
            
            result += '0' + temp % 2;
            carry = temp / 2;
        }
        reverse(result.begin(), result.end());
        return result;
    }
};

File: 0067-add-binary/0067-add-binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        length = max(len(a), len(b))
        if len(a) < length:
            a = '0' * (length - len(a)) + a
        if len(b) < length:
            b = '0' * (length - len(b)) + b
        
        carry = 0
        result = []
        for i in range(length - 1, -1, -1):
            if a[i] == '1':
                carry += 1
            if b[i] == '1':
                carry += 1
            
            result.append(str(carry % 2))
            carry //= 2
        
        if carry:
            result.append(str(carry))

        return "".join(result[::-1])